test/nmodl/transpiler/unit/CMakeLists.txt

```diff
@@ -207,20 +207,40 @@ endif()
 # =============================================================================
 if(NRN_ENABLE_PYTHON)
   if(NOT (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND NRN_SANITIZERS))
-    add_test(NAME Ode COMMAND ${PYTHON_EXECUTABLE} -m pytest ${CMAKE_CURRENT_SOURCE_DIR}/ode)
-    set_tests_properties(Ode PROPERTIES ENVIRONMENT "PYTHONPATH=${NMODL_TEST_PYTHONPATH}")
+    # Most of the Python tests added by `add_nrn_test` run in their own directories. Unfortunately,
+    # pytest-cov does not seem to have an option to specify the output file (it _always_ outputs to
+    # `coverage.xml`), which means that the various tests here may clobber each other's coverage
+    # reports. The way out of this is to a) either specify `WORKING_DIRECTORY` of each test, or b)
+    # use a coverage config file (see
+    # https://coverage.readthedocs.io/en/latest/config.html#xml-output); the latter requires far
+    # more work than the former, so we use option a) in the below.
+    file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/ode)
+    add_test(
+      NAME Ode
+      COMMAND ${PYTHON_EXECUTABLE} ${NRN_PYTEST_LAUNCHER} ${CMAKE_CURRENT_SOURCE_DIR}/ode
+      WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/ode)
+    set_tests_properties(
+      Ode
+      PROPERTIES ENVIRONMENT
+                 "PYTHONPATH=${NMODL_TEST_PYTHONPATH};NEURONHOME=${PROJECT_BINARY_DIR}/share/nrn")
     cpp_cc_configure_sanitizers(TEST Ode PRELOAD)
   endif()
 
   if(NMODL_ENABLE_PYTHON_BINDINGS)
     # Apple Clang and ASAN do not play along nicely with NMODL's Python bindings, so we skip these
     # tests
     if(NOT (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND NRN_SANITIZERS))
```

Comment:
any reason why the coverage would be gated behind the sanitizers? Took me a moment to figure out why I wasn't getting coverage, when I was looking at this.

Reply:
It's not the coverage that's gated, it's the tests themselves. TBH I had to jump through so many hurdles to get Apple Clang to compile even the most basic NEURON config, that I am not surprised there is guard here (and in a bunch of other CMake files). Usually any config with "advanced" features enabled (like sanitizers, or coverage, or backtrace) works as intended only on Linux.

Reply:
> It's not the coverage that's gated, it's the tests themselves. 

Sorry, yes, good point.
Perhaps it should be gated behind `NRN_ENABLE_COVERAGE` instead of `NRN_SANITIZERS`.